subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/result/DefaultComponentSelectionDescriptor.java

```diff
@@ -25,19 +25,30 @@
     private final Describable description;
     private final boolean hasCustomDescription;
     private final int hashCode;
+    private final boolean isForceAlignment;
 
     public DefaultComponentSelectionDescriptor(ComponentSelectionCause cause) {
         this.cause = cause;
         this.description = Describables.of(cause.getDefaultReason());
         this.hasCustomDescription = false;
         this.hashCode = cause.hashCode();
+        this.isForceAlignment = cause == ComponentSelectionCause.FORCED;
```

Comment:
:o: I'm not super excited by having a field here, but as a first step sounds ok.
:x: Should update `equals/hashCode`

Reply:
Yes, updating `equals`/ `hashCode`, my mistake